FDK/Tools/SharedData/FDKScripts/CheckOutlinesUFO.py

```diff
@@ -12,1020 +13,1145 @@
+        return None
+    glyph_name_list.append(glyph_name)
+
+    for glyphTag2 in range_list[1:]:
+        gid = get_glyph_id(glyphTag2, font_glyph_list)
+        if gid is None:
+            print("\tWarning: glyph ID <%s> in range %s from glyph selection "
+                  "list option is not in font. <%s>." %
+                  (glyphTag2, glyph_tag, font_file_name))
+            return None
+        for i in range(prev_gid + 1, gid + 1):
+            glyph_name_list.append(font_glyph_list[i])
+        prev_gid = gid
+
+    return glyph_name_list
+
+
+def filter_glyph_list(options, font_glyph_list, font_file_name):
+    # Return the list of glyphs which are in the intersection of the argument
+    # list and the glyphs in the font.
+    # Complain about glyphs in the argument list which are not in the font.
+    if not options.glyphList:
+        glyph_list = font_glyph_list
+    else:
+        # expand ranges:
+        glyph_list = []
+        for glyphTag in options.glyphList:
+            glyph_names = \
+                get_glyph_names(glyphTag, font_glyph_list, font_file_name)
+            if glyph_names is not None:
+                glyph_list.extend(glyph_names)
+    return glyph_list
+
+
+def get_digest(d_glyph):
+    """copied from robofab ObjectsBase.py.
+    """
+    mp = DigestPointPen()
+    d_glyph.drawPoints(mp)
+    digest = list(mp.getDigestPointsOnly(needSort=False))
+    digest.append((len(d_glyph.contours), 0))
+    return digest
+
+
+# noinspection PyProtectedMember
+def remove_coincident_points(b_glyph, changed, msg):
+    """ Remove coincident points.
+    # a point is (segment_type, pt, smooth, name).
+    # e.g. (u'curve', (138, -92), False, None)
+    """
+    for contour in b_glyph.contours:
+        op = contour._points[0]
+        i = 1
+        num_ops = len(contour._points)
+        while i < num_ops:
+            point = contour._points[i]
+            segment_type = point[0]
+            if segment_type is not None:
+                last_op = op
+                op = point
+                if last_op[1] == op[1]:
+                    # Coincident point. Remove it.
+                    msg.append("Coincident points at (%s,%s)." % (op[1]))
+                    changed = 1
+                    del contour._points[i]
+                    num_ops -= 1
+                    if segment_type == "curve":
+                        # Need to delete the previous two points as well.
+                        i -= 1
+                        del contour._points[i]
+                        i -= 1
+                        del contour._points[i]
+                        num_ops -= 2
+                    # Note that we do not increment i - this will
+                    # now index the next point.
+                    continue
+            i += 1
+    return changed, msg
+
+
+# noinspection PyProtectedMember
+def remove_tiny_sub_paths(b_glyph, min_area, changed, msg):
```

Comment:
`change` is not used within this function. Can it be removed?

Reply:
Yes. Will do.